merlin_models/tf/core.py

```diff
@@ -2270,14 +2337,137 @@ def test_step(self, inputs):
 
         return metrics
 
+    def fit(
+        self,
+        x=None,
+        y=None,
+        batch_size=None,
+        epochs=1,
+        verbose="auto",
+        callbacks=None,
+        validation_split=0.0,
+        validation_data=None,
+        shuffle=True,
+        class_weight=None,
+        sample_weight=None,
+        initial_epoch=0,
+        steps_per_epoch=None,
+        validation_steps=None,
+        validation_batch_size=None,
+        validation_freq=1,
+        max_queue_size=10,
+        workers=1,
+        use_multiprocessing=False,
+        **kwargs,
+    ):
+        # Check if merlin-dataset is passed
+        if hasattr(x, "to_ddf"):
+            if not batch_size:
+                raise ValueError("batch_size must be specified when using merlin-dataset.")
+            from .dataset import Dataset
+
+            x = Dataset(x, batch_size=batch_size, **kwargs)
+
+        return super().fit(
+            x,
+            y,
+            batch_size,
+            epochs,
+            verbose,
+            callbacks,
+            validation_split,
+            validation_data,
+            shuffle,
+            class_weight,
+            sample_weight,
+            initial_epoch,
+            steps_per_epoch,
+            validation_steps,
+            validation_batch_size,
+            validation_freq,
+            max_queue_size,
+            workers,
+            use_multiprocessing,
+        )
+
+    def batch_predict(self, dataset, batch_size=None, **kwargs):
+        if not set(self.schema.column_names).issubset(set(dataset.schema.column_names)):
+            raise ValueError(
+                f"Model schema {self.schema.column_names} does not match dataset schema"
+                + f" {dataset.column_names}"
+            )
+
+        # Check if merlin-dataset is passed
+        if hasattr(dataset, "to_ddf"):
+            dataset = dataset.to_ddf()
```

Comment:
I'm wondering if we should enforce that the `dataset` param is a merlin.io.Dataset class. It seems like if it isn't, the code on line 2394 will fail (since a dask ddf won't have a .schema member on it).

Reply:
Yeah you’re right! I added an extra check for this. I just want to avoid checking with `isinstance` since we could get a `nvt.Dataset` as well.